File: leetcode/2024_6_june/python/10_height_checker.py
# CODE

from typing import List
class Solution:
    def heightChecker(self, heights: List[int]) -> int:
        org_heights = heights.copy()
        heights.sort()
        res = 0
        for i in range(len(org_heights)):
            if org_heights[i] != heights[i]:
                res += 1
        return res

File: leetcode/2024_6_june/c++/10_height_checker.cpp
// TIME COMPLEXITY
// O(nlogn)

// SPACE COMPLEXITY
// O(n)

// CODE

class Solution {
public:
    int heightChecker(vector<int>& heights) {
        vector<int> sorted(heights.begin(), heights.end());
        int cnt = 0;
        sort(sorted.begin(), sorted.end());
        for(int i = 0; i < heights.size(); i++){
            if(heights[i] != sorted[i]){
                cnt++;
            }
        }
        return cnt;
        
    }
};
